pkg/conn/basedb.go

```diff
@@ -56,10 +56,16 @@ func (d *DefaultDBProviderImpl) Apply(config config.DBConfig) (*BaseDB, error) {
 		}
 		maxIdleConns = rawCfg.MaxIdleConns
 	}
+
+	for key, val := range config.Session {
+		dsn += fmt.Sprintf("&%s=\"%s\"", key, val)
```

Comment:
Do we need to use different formats for types of `val`? see https://github.com/go-sql-driver/mysql#system-variables

Reply:
Because we don't restrict user config, it's hard to use different formats between int and string.
I use `'` for all types and url.QueryEscape to escape string in 9ec51cb. MySQL extracts the relevant type from a string.

Reply:
can it work if add "" for number value?

Reply:
yes，see  https://github.com/pingcap/dm/blob/9ec51cb2ace7e31d8c61112134f5b0a795bf938f/tests/all_mode/conf/dm-task.yaml#L20

Reply:
> MySQL extracts the relevant type from a string.

Are there any docs/links explain about this?

Reply:
Seems not official. https://github.com/go-sql-driver/mysql/issues/405#issuecomment-256257156

Reply:
👍 you can add this as a comment above the code.

Reply:
add commet in 5d57a5b